packages/schematics/angular/utility/dependencies.ts

```diff
@@ -0,0 +1,153 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { JsonAstObject, JsonParseMode, parseJsonAst } from '@angular-devkit/core';
+import { SchematicsException, Tree } from '@angular-devkit/schematics';
+import {
+  appendPropertyInAstObject,
+  findPropertyInAstObject,
+} from './json-utils';
+
+
+const pkgJsonPath = '/package.json';
+export enum DependencyType {
+  Default,
+  Dev,
+  Peer,
+  Optional,
+}
+
+export interface Dependency {
+  type: DependencyType;
+  packageName: string;
+  version: string;
+  overwrite?: boolean;
+}
+
+export function addPackageJsonDependencies(tree: Tree, dependencies: Dependency[]): void {
+    dependencies
+      .forEach(dependency => addPackageJsonDependency(tree, dependency));
+}
+
+export function addPackageJsonDependency(tree: Tree, dependency: Dependency): void {
+  const packageJsonAst = _readPackageJson(tree);
+  const dependencyPropName = _getDependencyPropName(dependency.type);
+  const depsNode = findPropertyInAstObject(packageJsonAst, dependencyPropName);
+  const recorder = tree.beginUpdate(pkgJsonPath);
+
+  if (!depsNode) {
+    // Haven't found the dependencies key, add it to the root of the package.json.
+    appendPropertyInAstObject(recorder, packageJsonAst, dependencyPropName, {
+      [dependency.packageName]: dependency.version,
+    });
+  } else if (depsNode.kind === 'object') {
+    // check if package already added
+    const depNode = findPropertyInAstObject(depsNode, dependency.packageName);
+
+    if (!depNode) {
+      // No build-angular package, add it.
+      appendPropertyInAstObject(
```

Comment:
These need to be sorted. I don't think this is doing it.

Reply:
The sorting logic is below, anytime a package is added that node is sorted.

Reply:
Instead of using `findPropertyInAstObject`, the function could iterate through the keys and determine both the sorted insertion point and whether it already exists at the same time.

Reply:
If it is already out of order then this can't really make it any worse.  And if the file is purposefully out of order, should this really change that (non-)order?  Ideally the diff should be minimal either way.